java-frontend/src/main/java/org/sonar/java/collections/PStack.java

```diff
@@ -41,6 +43,14 @@
   E peek();
 
   /**
+   *
+   * @param i - index of element to be returned, 0 means top of the stack
+   * @return i-th element from top of the stack or null, if stack doesn't have so many elements
+   */
+  @CheckForNull
+  E peek(int i);
```

Comment:
I would prefer to have the method throw an `IllegalStateException` if we try to access something out of the stack. When using this method, it means that you perfectly know what you are doing, trying to access a given element from the stack. If you can't I would prefer to have it failing fast.